src/filters/mtls.rs

```diff
@@ -0,0 +1,52 @@
+//! Mutual (client) TLS filters.
+
+use std::convert::Infallible;
+
+use rustls_pki_types::CertificateDer;
+
+use crate::{
+    filter::{filter_fn_one, Filter},
+    route::Route,
+};
+
+/// Certificates is a iterable container of Certificates.
+pub type Certificates = Vec<CertificateDer<'static>>;
+
+/// Creates a `Filter` to get the peer certificates for the TLS connection.
+///
+/// If the underlying transport doesn't have peer certificates, this will yield
+/// `None`.
+///
+/// # Example
+///
+/// ```
+/// use warp::mtls::Certificates;
+/// use warp::Filter;
+///
+/// let route = warp::mtls::peer_certificates()
+///     .map(|certs: Option<Certificates>| {
+///         println!("peer certificates = {:?}", certs.as_ref());
+///     });
+/// ```
+pub fn peer_certificates(
+) -> impl Filter<Extract = (Option<Certificates>,), Error = Infallible> + Copy {
+    filter_fn_one(|route| futures_util::future::ok(from_route(route)))
+}
+
+/// Testing
```

Comment:
This comment suggests that this function is just for testing purposes. There are other PRs in `warp` to implement mtls that do not use it at all. The purpose of this func is to return a owned value for some reason, which is easier to manage, but not really needed in prod code. I think we may just discard it.